pennylane/resource/first_quantization.py

```diff
@@ -87,19 +88,39 @@ def __init__(
         error=0.0016,
         charge=0,
         br=7,
+        vectors=None,
     ):
         self.n = n
         self.eta = eta
         self.omega = omega
         self.error = error
         self.charge = charge
         self.br = br
+        self.vectors = vectors
+        self.cubic = True
+
+        if self.vectors is not None:
+            recip_v = (
+                2
+                * np.pi
+                / np.abs(np.sum((np.cross(vectors[0], vectors[1]) * vectors[2])))
+                * np.array([np.cross(vectors[i], vectors[j]) for i, j in [(1, 2), (2, 0), (0, 1)]])
+            )
+            bbt = np.matrix(recip_v) @ np.matrix(recip_v).T
+            cubic = np.linalg.norm(bbt - (recip_v**2).max() * np.identity(3)) < 1e-6
+            if not cubic:
+                self.cubic = False
```

Comment:
Do we have a reference for computing this? e.g., a book or a paper.

Reply:
For computing the reciprocal lattice vectors, I guess we can reference Eq. (35) of [arXiv:2302.07981](https://arxiv.org/pdf/2302.07981.pdf), but I am sure any basic quantum chemistry book talking about lattices would do.

For the equations after that, computing the cubicity/(non)orthogonality, that's just standard linear algebra following the definition of cubic/(non)orthogonal, so no reference is needed.